sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1352,7 +1352,11 @@ object CodeGenerator extends Logging {
       }
     }.flatten
 
-    codeSizes.max
+    if (codeSizes.nonEmpty) {
+      codeSizes.max
+    } else {
+      0
```

Comment:
Do we have a test case? cc @rednaxelafx

Reply:
Currently, no. I tried to add tests though, I couldn't find an input for this issue... Any idea, Kris?

Reply:
my 2 cents - We only have this issue on our most heavy prod jobs with a lot of columns and a lot of partitions.. I wonder if Spark struggles to collect some stats in this case, and then this becomes zero. We don't have a reproducer as it again only happens on our heaviest prod jobs.. but it does happen consistently when we're over a certain limit of number of columns or number of partitions/tasks, or both. Thanks.